api/nodes/views.py

```diff
@@ -92,17 +97,56 @@ class NodeDetail(generics.RetrieveUpdateAPIView, NodeMixin):
         ContributorOrPublic,
         ReadOnlyIfRegistration,
     )
+
     serializer_class = NodeSerializer
 
-    # overrides RetrieveUpdateAPIView
+    # overrides RetrieveUpdateDestroyAPIView
     def get_object(self):
         return self.get_node()
 
-    # overrides RetrieveUpdateAPIView
+    # overrides RetrieveUpdateDestroyAPIView
     def get_serializer_context(self):
         # Serializer needs the request in order to make an update to privacy
         return {'request': self.request}
 
+    # overrides RetrieveUpdateDestroyAPIView
+    def perform_destroy(self, instance):
+        user = self.request.user
+        node = self.get_node()
+        token = node_token_creator(node._id, user._id)
+        url = absolute_reverse('nodes:node-delete-confirm', kwargs={'node_id': node._id, 'token': token})
+        delete_warning = BEFORE_DELETE_NODE.format(_(node.title))
+        raise Accepted([delete_warning, url])
```

Comment:
It feels very weird to me to raise an exception in the default case. Could we not return a response with the correct status code, without raising it as an exception?